packages/compass-indexes/src/components/search-index-template-dropdown/index.tsx

```diff
@@ -0,0 +1,74 @@
+import { ATLAS_SEARCH_TEMPLATES } from '@mongodb-js/mongodb-constants';
+import type { SearchTemplate } from '@mongodb-js/mongodb-constants';
+import React, { useState, useCallback } from 'react';
+import {
+  Select,
+  Option,
+  Icon,
+  css,
+  spacing,
+  Tooltip,
+} from '@mongodb-js/compass-components';
+
+const dropdownStyles = css({
+  display: 'flex',
+  overflow: 'hidden',
+  gap: 0,
+});
+
+const overlapDropdownLabelStyles = css({
+  marginLeft: -spacing[3],
+});
```

Comment:
This is a very unconventional way of positioning this icon, flex is not a good way to do it and you are hacking on top of it to align the icon inside the label. While leafygreen types only allow to pass strings to the label, you can actually pass components there, or, if we don't want to do that, leafygreen label is not a good match here and you need to add your own label with an icon inside

Reply:
Yeah I'm not sure about this solution either, but the design adds an information icon to a component that doesn't have it. So I isolated it in a single component in case we want to change it in the future or not and implemented the easiest way that shouldn't break the layout (tested in different sizes of the popup).

I can try passing a component for the label itself, that has the flex config inside, and see if it works properly.